cpp/tests/types/types_test.cpp

```diff
@@ -209,8 +213,8 @@ TYPED_TEST(WrappersNoBoolTest, BooleanOperators)
 }
 
 // Ensure bool8 constructor is correctly casting the input type to a bool
-TEST_F(WrappersTestBool8, BooleanConstructor) {
-  using UnderlyingType = cudf::bool8::value_type;
+TYPED_TEST(WrappersBoolTest, BooleanConstructor) {
```

Comment:
Existing tests shouldn't need to be modified. Furthermore, this isn't really testing what we want.

We want a new `TYPED_TEST_CASE` that is tested across all the types `int8_t, int16_t, ... float, double` to ensure that constructing a `bool8` from those types will properly convert to a `bool` and not suffer from overflow.